test/exposure.jl

```diff
@@ -0,0 +1,129 @@
+    
+    ruleTreatment = Threshold_Exposure{Tuple{:pesticide,:population}, :pesticide}(crop=Aux(:cropDD), pulseLevel=20, popThreshold = 1000)
+
+    rule = Ruleset(ruleTreatment)
+    sim!(output, rule)
+    @test output[Ti(1)] == (
+        pesticide = [0.0 0.0 0.0; 0.0 0.0 0.0; 0.0 0.0 0.0],
+        population = [100000.0 100.0 0.0; 0.0 1000.0 0.0; 0.0 0.0 100000.0])
+    @test output[Ti(2)] == (
+        pesticide = [0.0 0.0 0.0; 0.0 20.0 0.0; 0.0 0.0 20.0],
+        population = [100000.0 100.0 0.0; 0.0 1000.0 0.0; 0.0 0.0 100000.0])
+    @test output[Ti(3)] == (
+        pesticide = [0.0 0.0 0.0; 0.0 32.13061319425267 0.0; 0.0 0.0 32.13061319425267],
+        population = [100000.0 100.0 0.0; 0.0 1000.0 0.0; 0.0 0.0 100000.0])
+
+end   
+
+##################
+# GENERIC ROTATION
+##################
+
+@testset "rotation structure" begin
+    RS(x,y) = Dispersal.RotationStruct(x,y)
+   
+    @test mod(RS(1,2),2) == mod(1,2)
+    @test mod(RS(2,2),2) == mod(2,2)
+    @test mod(RS(2,3),2) == mod(2,2)
+    @test mod(RS(3,2),2) == mod(3,2)
+
+    @test RS(3,2) + 1 == RS(4,2)
+    @test RS(3,2) + 8.3 == RS(11.3,2)
+
+    US(x,y) = Dispersal.updateStep(x, y)
+    @test US(RS(3,2), 4) == RS(3,4)
+    @test US(RS(3,DateTime(2020,2,1)), DateTime(2021,1,10)) == RS(3,DateTime(2021,1,10))
+
+    @test Rotation_InitGrid(1, (3,3)) == [
+        RS(1,1) RS(1,1) RS(1,1);
+        RS(1,1) RS(1,1) RS(1,1);
+        RS(1,1) RS(1,1) RS(1,1)
+    ]
+
+    @test Rotation_InitGrid(DateTime(2020,2,1), (2,2)) == [
+        RS(1,DateTime(2020,2,1)) RS(1,DateTime(2020,2,1)) ;
+        RS(1,DateTime(2020,2,1)) RS(1,DateTime(2020,2,1))
+    ]
+end
+
+
+@testset "Test rotation in exposure" begin
+    nr, nc = 1, 1
+    init = (
+        pesticide1 = zeros(nr,nc),
+        pesticide2 = zeros(nr,nc),
+        pesticide3 = zeros(nr,nc),
+        rotation = Rotation_InitGrid(1, (nr,nc)),
+        population = ones(nr,nc).*10^4, )
+
+    crop = ones(nr,nc)
+       
+    output = ArrayOutput(init; tspan=1:7,  aux=(crop=crop,))
+    ruleTreatment1 = Rotation_Exposure{Tuple{:pesticide1,:population,:rotation}, Tuple{:pesticide1,:rotation}}(
+        crop=Aux(:crop), rotationSize=3, rotationNumber= 1, pulseLevel=20.0, popThreshold = 1000.0, degradationRate=0.5
+    )
+    ruleTreatment2 = Rotation_Exposure{Tuple{:pesticide2,:population,:rotation}, Tuple{:pesticide2,:rotation}}(
+        crop=Aux(:crop), rotationSize=3, rotationNumber= 2, pulseLevel=20.0, popThreshold = 1000.0,degradationRate=0.5
+    )
+    ruleTreatment3 = Rotation_Exposure{Tuple{:pesticide3,:population,:rotation}, Tuple{:pesticide3,:rotation}}(
+        crop=Aux(:crop), rotationSize=3, rotationNumber= 3, pulseLevel=20.0, popThreshold = 1000.0,degradationRate=0.5
+    )
+    rule = Ruleset(Chain(ruleTreatment1, ruleTreatment2,ruleTreatment3))
+    sim!(output, rule)
+
+    CheckReturn(i) = [output[i].pesticide1[1] , output[i].pesticide2[1], output[i].pesticide3[1]]
```

Comment:
functions are lower case/snake case: `checkreturn(i)`. This looks like a type constructor to me.